pennylane/labs/resource_estimation/resource_tracking.py

```diff
@@ -90,14 +90,21 @@ def resources_from_operation(
 ) -> Resources:
     """Get resources from an operation"""
 
-    if isinstance(obj, ResourceConstructor):
-        cp_rep = obj.resource_rep_from_op()
-        gate_counts_dict = defaultdict(int)
-        _counts_from_compressed_res_op(cp_rep, gate_counts_dict, gate_set=gate_set, config=config)
-        return Resources(gate_types=gate_counts_dict)
+    if not isinstance(obj, Operation):
+        raise TypeError(f"obj must be of type Operation. Got type {type(obj)}.")
+
+    if not isinstance(obj, ResourceOperator):
+        obj = _op_to_resource_op(obj)
+
+    cp_rep = obj.resource_rep_from_op()
+    gate_counts_dict = defaultdict(int)
+    _counts_from_compressed_res_op(cp_rep, gate_counts_dict, gate_set=gate_set, config=config)
+
+    num_wires = len(obj.wires)
+    num_gates = sum(gate_counts_dict.values())
+
+    return Resources(gate_types=gate_counts_dict, num_gates=num_gates, num_wires=num_wires)
```

Comment:
Awesome, thanks for implementing this! I will leave the testing of this piece to you as well. Although I will be adding tests soon for `get_resources`, you should be able to copy the test and just change parts of it to test this sub-case.

Reply:
I'll let you know once I add it.